pallets/msa/src/lib.rs

```diff
@@ -397,8 +413,55 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_initialize(current: T::BlockNumber) -> Weight {
+			<MSAEventCount<T>>::set(0u16);
+			// allocates 1 read and 1 write for any access of `MSAEventCount` in every block
+			T::DbWeight::get().reads(1u64).saturating_add(T::DbWeight::get().writes(1u64));
```

Comment:
quickly referencing from messages pallet to use this counter type  to bridge number of unique events indexed in a given block.
* Assuming offchain_workers will see their respective MSAEventCount, as the block_number they are processing?   
* If not do we really save more than just a Counter but also which block_number that counter is for?
@aramikm

Reply:
Looks great!